### PR TITLE
refactoring and add proxy arg

### PR DESCRIPTION
**Pull request by dc-avasilev**
close:
https://github.com/testit-tms/autotest-integration-python/issues/10
https://github.com/testit-tms/autotest-integration-python/issues/2

### CODE REVIEW

**Review by testit-tms — CHANGES_REQUESTED** [review]

File: testit_pytest/src/listener.py [hunk 1]
```diff
@@ -39,49 +43,78 @@ def __init__(self, testrun, url, private_token):
...
-            if hasattr(item.function, 'test_externalID'):
-                if item.own_markers:
-                    for mark in item.own_markers:
-                        if mark.name == 'parametrize':
-                            if not hasattr(item, 'array_parametrize_id'):
-                                item.array_parametrize_id = []
-                            item.array_parametrize_id.append(item.own_markers.index(mark))
-
-                item.test_externalID = self.param_attribute_collector(item.function.test_externalID, item.own_markers, item.array_parametrize_id, index) if hasattr(item, 'array_parametrize_id') else item.function.test_externalID
+            if hasattr(item.function, 'test_externalID') and item.own_markers:
```
[the remark this answers is not in the document]
Comment (dc-avasilev) [reply testit_pytest/src/listener.py:89]: this is your old code, I did not change the logic, I just optimized it visually for easier perception, what do you want me to do here? I can return everything as it was with you =)

[the remark this answers is not in the document]
Comment (dc-avasilev) [reply testit_pytest/src/listener.py:89]: oops sorry I saw that the indentation moved out and that the condition levels changed

[the remark this answers is not in the document]
Comment (dc-avasilev) [reply testit_pytest/src/listener.py:89]: ++
repaired

File: testit_pytest/src/plugin.py [hunk 2]
```diff
@@ -30,10 +30,20 @@ def pytest_addoption(parser):
         metavar="T2lKd2pLZGI4WHRhaVZUejNl",
         help='Set API secret key'
     )
+    parser.getgroup('debug').addoption(
```
[the remark this answers is not in the document]
Comment (dc-avasilev) [reply testit_pytest/src/plugin.py:33]: ++
repaired

File: README.rst [hunk 3]
```diff
@@ -61,15 +63,16 @@ Usage with a connection_config.ini file in the root directory of the project:
...
+    # testit_proxy = {"http": "http://localhost:8888", "https": "http://localhost:8888"}
 The second option
 ------------------
 
 Usage with command-line parameters in a CI/CD file:
 
 .. code::
 
-    pytest --testit --testrunid=${TEST_RUN_ID} --url=${URL} --privatetoken=${PRIVATE_TOKEN}
+    pytest --testit --testrunid=${TEST_RUN_ID} --url=${URL} --privatetoken=${PRIVATE_TOKEN} --testit_proxy='{"http":"http://localhost:8888","https":"http://localhost:8888"}'
```
[the remark this answers is not in the document]
Comment (dc-avasilev) [reply README.rst:75]: ++
repaired

**Pull request merged**